cfn-resources/teams/cmd/resource/team-user/team_user_test.go

```diff
@@ -0,0 +1,267 @@
+	}
+
+	// Check that the returned slice contains only the valid usernames
+	assert.Equal(t, len(validUsers), 2)
+	assert.Equal(t, validuser1, *validUsers[0].Id)
+	assert.Equal(t, validuser2, *validUsers[1].Id)
+}
+
+func TestValidateUsernamesWithInvalidInput(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+
+	mockAtlasV2Client := mock.NewMockTeamUsersAPI(mockCtrl)
+
+	// Create a slice of usernames, including some valid and some invalid usernames
+	validuser1 := "validuser1"
+	invaliduser1 := "invaliduser1"
+	usernames := []string{validuser1, invaliduser1}
+
+	// Set up mock expectations for successful and unsuccessful calls to GetUserByUsername
+	mockAtlasV2Client.EXPECT().GetUserByUsername(context.Background(), validuser1).Return(&atlasv2.CloudAppUser{Id: &validuser1}, nil, nil)
+	mockAtlasV2Client.EXPECT().GetUserByUsername(context.Background(), invaliduser1).Return(nil, nil, errors.New("invalid username"))
+
+	// Call the ValidateUsernames function
+	_, _, err := ValidateUsernames(mockAtlasV2Client, usernames)
+
+	assert.NotNil(t, err)
+}
+
+func TestGetUsersToAddAndRemove(t *testing.T) {
+	user1 := "user1"
+	user2 := "user2"
+	user3 := "user3"
+
+	// Test cases
+	testCases := []struct {
+		testName         string
+		currentUsers     []atlasv2.CloudAppUser
+		newUsers         []atlasv2.CloudAppUser
+		expectedToAdd    []string
+		expectedToDelete []string
+	}{
+		{
+			testName: "succeeds adding a new user and removing an existing one",
+			currentUsers: []atlasv2.CloudAppUser{
+				{Id: &user1},
+				{Id: &user2},
+			},
+			newUsers: []atlasv2.CloudAppUser{
+				{Id: &user1},
+				{Id: &user3},
+			},
+			expectedToAdd:    []string{user3},
+			expectedToDelete: []string{user2},
+		},
+		{
+			testName:     "succeeds adding all users",
+			currentUsers: []atlasv2.CloudAppUser{},
+			newUsers: []atlasv2.CloudAppUser{
+				{Id: &user1},
+				{Id: &user2},
+			},
+			expectedToAdd:    []string{user1, user2},
+			expectedToDelete: []string{},
+		},
+		{
+			testName: "succeeds removing both users",
+			currentUsers: []atlasv2.CloudAppUser{
+				{Id: &user1},
+				{Id: &user2},
+			},
+			newUsers:         []atlasv2.CloudAppUser{},
+			expectedToAdd:    []string{},
+			expectedToDelete: []string{user1, user2},
+		},
+	}
+
+	// Run test cases
+	for _, testCase := range testCases {
+		t.Run(testCase.testName, func(t *testing.T) {
+			toAdd, toDelete, err := GetTeamUserUpdates(testCase.currentUsers, testCase.newUsers)
+			require.NoError(t, err)
+			assert.Equal(t, testCase.expectedToAdd, toAdd)
+			assert.Equal(t, testCase.expectedToDelete, toDelete)
+		})
+	}
+}
+
+func TestUpdateTeamUsers(t *testing.T) {
+	validuser1 := "validuser1"
+	validuser2 := "validuser2"
+	invaliduser1 := "invaliduser1"
+
+	testCases := []struct {
+		testName             string
+		mockFuncExpectations func(*gomock.Controller) *mock.MockTeamUsersAPI
+		existingTeamUsers    *atlasv2.PaginatedApiAppUser
+		usernames            []string
+		expectError          bool
```

Comment:
```suggestion
		wantErr require.ErrorAssertionFunc
```
https://pkg.go.dev/github.com/stretchr/testify@v1.8.4/assert#ErrorAssertionFunc